client/components/advanced-credentials/index.tsx

```diff
@@ -89,17 +90,32 @@ const AdvancedCredentials: FunctionComponent< Props > = ( { action, host, role }
 	const { protocol } = credentials;
 	const isAtomic = hasCredentials && 'dynamic-ssh' === protocol;
 
+	const [ testCredentialsResult, setTestCredentialsResult ] = useState( false );
+	const [ testCredentialsLoading, setTestCredentialsLoading ] = useState( true );
+
+	useEffect( () => {
+		( async () => {
+			const results: { ok: boolean } = await wp.req.post( {
+				path: '/sites/' + siteId + '/rewind/credentials/test?role=main',
+				apiNamespace: 'wpcom/v2',
+			} );
+			const { ok } = results;
+			setTestCredentialsLoading( false );
+			setTestCredentialsResult( ok );
+		} )();
+	}, [] );
+
 	const statusState = useMemo( (): StatusState => {
-		if ( isRequestingCredentials ) {
+		if ( isRequestingCredentials || testCredentialsLoading ) {
```

Comment:
Should we put `testCredentialsLoading` right in the `isRequestingCredentials` selector function? Because right now someone could accidentally use `isRequestingCredentials` and miss `testCredentialsLoading`.

Reply:
Done, thanks for the suggestion! I think this does make it a bit cleaner and more efficient.